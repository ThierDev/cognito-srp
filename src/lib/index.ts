export * from './ClientPasswordChallenge';
export * from './ServerPasswordChallenge';
export * from './Session';
export * from './UserPool';
