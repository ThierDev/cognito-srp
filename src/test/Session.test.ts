import { Session } from '../lib';
import {
  poolname,
  clientUser,
  sessionKey,
  scramblingParameter,
  hkdf
} from './constants';

describe('Session', () => {
  it('should be constructable from session key and scrambling parameter', () => {
    const session = new Session(
      poolname,
      clientUser.username,
      Buffer.from(sessionKey, 'hex'),
      Buffer.from(scramblingParameter, 'hex')
    );
    expect(session.getHkdf()).toMatchSnapshot();
    expect(session.calculateSignature('secret', 'timestamp')).toMatchSnapshot();
  });

  it('should be constructable from hkdf', () => {
    const session = new Session(poolname, clientUser.username, hkdf);
    expect(session.getHkdf()).toMatchSnapshot();
    expect(session.calculateSignature('secret', 'timestamp')).toMatchSnapshot();
  });
});
