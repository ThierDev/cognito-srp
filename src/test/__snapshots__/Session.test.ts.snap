// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session should be constructable from hkdf 1`] = `"81bea5ea37891c30ee91334d166776ad"`;

exports[`Session should be constructable from hkdf 2`] = `"hu5kSl9dOiew86nl0jk/Q6AvwWOXeOLtEFNtUdtmiMk="`;

exports[`Session should be constructable from session key and scrambling parameter 1`] = `"81bea5ea37891c30ee91334d166776ad"`;

exports[`Session should be constructable from session key and scrambling parameter 2`] = `"hu5kSl9dOiew86nl0jk/Q6AvwWOXeOLtEFNtUdtmiMk="`;
